lib/order.rb

```diff
@@ -0,0 +1,34 @@
+require 'confirmation_text_sender'
+
+class Order
+
+  attr_reader :order, :total
+
+  def initialize(order, text_confirm_order = ConfirmationTextSender.new)
+    @order = order
+    @total = 0
+    @text_confirm_order = text_confirm_order
+  end
+
+  def confirm_order
```

Comment:
You have three methods here that all have names starting `confirm_` but they have very different behaviour.  This makes the interface confusing.